packages/framework/src/commands/CommandManager.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  APIApplicationCommand,
+  ApplicationCommandType,
+  RESTGetAPIApplicationCommandsResult,
+  RESTPatchAPIApplicationCommandJSONBody,
+  RESTPatchAPIApplicationCommandResult,
+  RESTPostAPIApplicationCommandsJSONBody,
+  RESTPostAPIApplicationCommandsResult,
+  Routes,
+} from "discord-api-types/v10";
+import type { App } from "../client";
+import { ChatInputRoute } from "../router";
+import type { Command } from "./Command";
+
+export class CommandManager {
+  public constructor(private app: App) {}
+
+  private readonly commands = new Map<string, Command>();
+
+  public registerCommand(command: Command) {
+    this.app.router.addRoute(new ChatInputRoute(this.app, command));
+    this.commands.set(command.options.name, command);
+  }
+
+  /**
+   * Sync all registered commands with the DAPI
+   * @param guildId Guild ID to sync commands for
+   */
+  public async syncCommands(guildId?: string) {
+    const existingCommands = await this.getRegisteredCommands({
+      guildId,
+      onlyFramework: true,
+    });
+
+    const commandsToUpdate = this.filterExistingCommands(existingCommands);
+
+    this.app.logger.debug(
+      `Found ${commandsToUpdate.length} commands to update!`,
+      commandsToUpdate
+    );
+
+    const commandsToRegister =
+      this.filterNonFrameworkCommands(existingCommands);
+
+    this.app.logger.debug(
+      `Found ${commandsToRegister.length} commands to register!`,
+      commandsToRegister
+    );
+
+    for (const command of commandsToRegister) {
+      await this.registerCommandToDiscord(command, guildId);
+    }
+
+    for (const [command, existingCommand] of commandsToUpdate) {
+      await this.updateCommandOnDiscord(existingCommand.id, command, guildId);
+    }
```

Comment:
```suggestion  
    await Promise.all(commandsToRegister.map(async (command) => {
    	await this.registerCommandToDiscord(command, guildId);
    }));
    
    await Promise.all(commandsToUpdate.map(([command, existingCommand]) => {
      await this.updateCommandOnDiscord(existingCommand.id, command, guildId);
    }));
```

Reply:
resolved in https://github.com/Disploy/disploy/pull/19/commits/48d704364b608c1fabe85ee345461d70cf1f76a1